### PR TITLE
Remove allow_hosts as it it no longer supported

### PR DESCRIPTION
**Pull request by gnuoy**
See https://github.com/juju-solutions/layer-basic/issues/149 for more details

### CODE REVIEW

**Comment by ajkavanagh** [comment]
Just hit this on focal with a octavia-dashboard charm (testing for focal readiness).  We need to decide whether this is going in, or find an alternative, soon, please.

**Comment by fnordahl** [comment]
I'm hitting this with the OVN charms too.

> If it's safe to assume that pypa/pip#410 is fixed, I think we should just drop the entire file. I'm a little concerned that this might reintroduce juju/charm-tools#117 on older instances (trusty, I think), but maybe that's moot since we also bundle a newer pip in the wheelhouse and install that first?

@johnsca by "drop the entire file" do you refer to removing this section?:
https://github.com/juju-solutions/layer-basic/blob/f8629dbe4eba480677eb613f218f8a0a384709e4/lib/charms/layer/basic.py#L127-L134

**Comment by johnsca** [comment]
@fnordahl Leaving the file won't hurt anything but the whole point of putting it in place in the first place was for the `allow_hosts` line to stop the bootstrap phase from trying (and failing) to make a request to PyPI in a network restricted environment.  As long as we're confident that installing a no-op reactive charm (e.g., cs:ubuntu) on an instance that can't talk to PyPI still works on trusty, I'm ok with merging as-is.

File: lib/charms/layer/basic.py [hunk 1]
```diff
@@ -69,7 +69,6 @@ def bootstrap_charm_deps():
             # (see https://github.com/pypa/pip/issues/410)
             fp.writelines([
                 "[easy_install]\n",
-                "allow_hosts = ''\n",
                 "find_links = file://{}/wheelhouse/\n".format(charm_dir),
             ])
```
Comment (johnsca) [inline lib/charms/layer/basic.py:73]: What about this instead:

```suggestion
        if series in ('ubuntu12.04', 'precise',
                          'ubuntu14.04', 'trusty',
                          'ubuntu16.04', 'xenial',
                          'ubuntu18.04', 'bionic'):
            with open('/root/.pydistutils.cfg', 'w') as fp:
                # make sure that easy_install also only uses the wheelhouse
                # (see https://github.com/pypa/pip/issues/410)
                fp.writelines([
                    "[easy_install]\n",
                    "allow_hosts = ''\n",
                    "find_links = file://{}/wheelhouse/\n".format(charm_dir),
                ])```

**Comment by johnsca** [comment]
Closing in favor of #169

**Pull request closed without merging**